### PR TITLE
(huge) refactor codebase

### PR DESCRIPTION
**Pull request by faultables**
So I just doing _some_ refactor

## presentational & container component

almost all component in fragments I rewrite it to class-based component.

well, not actually. But mostly, yes. Some fragments that I think will have more complex component lifecycle I rewrited it to class-based components. The big difference between `fragments` and `components` were `components` mostly only for "displaying" data. maybe there are some local state with React hooks help, but if you want to deal with service, it should be done in `fragments` not `components`.

Except, some component that doesn't have complex data flow.

## <AuthHoc />

Now all "protected" route are wrapped by `AuthHoc` (thanks to @ri7nz), as long as I test it manually, it works seamlessly.

## test scenarios

some updates here are mostly just adding some new file for test purpose. For `fragments` and `components` test scenario, it just some basic test and mostly I write some mock for the props.

this is the current state of our test

Screen Shot 2019-06-29 at 07 56 23

will write more test until ~90% code coverage

## async/await tradeoffs

I have love-hate relationship with `async/await`, and for me its hard too test. Also, our app was very rely on `EventEmitter`, still learning more about how to avoid memory leak in this app and let the test runner running smoothly without unpredictable result

---

There are so many updates related this commit, but I give me some time to remember it, thanks.

### CODE REVIEW

**Comment by anakmagang** [comment]
Current deployment URL: https://nadya-app-canary-as32ha5zl.now.sh

**Comment by anakmagang** [comment]
Current deployment URL: https://nadya-app-canary-o4531hffk.now.sh

**Comment by anakmagang** [comment]
Current deployment URL: https://nadya-app-canary-ja5k004db.now.sh

**Comment by anakmagang** [comment]
Current deployment URL: https://nadya-app-canary-ja5k004db.now.sh

**Comment by anakmagang** [comment]
Current deployment URL: https://nadya-app-canary-312p6aytk.now.sh

**Comment by faultables** [comment]
current bundle graph (look at that fucking simple-icons!). maybe we can reduce it by code splitting it, or something else

Screen Shot 2019-06-29 at 08 36 47

**Comment by anakmagang** [comment]
Current deployment URL: https://nadya-app-canary-8h6z6fz77.now.sh

**Comment by r17x** [comment]
Please make an issue for material-icons (separately) @108kb

**Comment by faultables** [comment]
Material icons or simple icons?

File: app/src/components/Detail.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  test('It should render Detail screen', () => {
+    const subscription = {
+      title: 'Spotify'
+    }
+    const handleDelete = () => {}
+    const { container } = render(
+      <Detail handleDelete={handleDelete} subscription={subscription} />
+    )
+
+    expect(container).toBeInTheDocument()
```
Comment (r17x) [inline app/src/components/Detail.test.js:17]: good job, we can add more assessment, if you have a state or value was render (show for user).
expect(assertTarget).toBeVisible()

**Review by r17x — APPROVED** [review]

File: app/src/components/Detail.test.js (hunk 1 above)
Comment (faultables) [reply app/src/components/Detail.test.js:17]: Thanks for review, will merge it 🚀

**Pull request merged**